docs/develop/natmod.rst

```diff
@@ -67,6 +67,14 @@ The known limitations are:
 So, if your C code has writable data, make sure the data is defined globally,
 without an initialiser, and only written to within functions.
 
+Linker limitation: the native module is not linked against the symbol table of the
+full micropython firmware. Rather, it is linked against an explicit table of exported
+symbols found in ``mp_fun_table``. It is thus not possible to simply call some arbitrary function
+in FreeRTOS, for example. Adding new symbols to the end of the table is relatively easy,
```

Comment:
I'd rephrase this to be a bit more general, eg something like "... some arbitrary HAL/OS/RTOS/system function, for exampe".